### PR TITLE
Refactor

### LINKED ISSUE #6: Rename Config file to Environments file

### PR DESCRIPTION
**Pull request by prog893**
closes: #6 

- Better help
- More safety checks on `eden config check`
- Refactor DynamoDB controller into class (will later be moved to core)
- Fix push/pull bugs with new schema
- Fix ls issues

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,14 +201,15 @@ $ eden config push -p api
 Successfully pushed profile api to DynamoDB table eden
 ```
 
-Use remote-rm to remove remote profiles
+Use remote-rm to remove remote profiles:
```
Comment (toVersus) [inline README.md]:
```suggestion
Use remote-rm to delete remote profiles:
```

File: aws_eden_cli/dynamodb.py [hunk 2]
```diff
@@ -4,262 +4,292 @@
...
+            table_status = self.describe_remote_state_table()
+        except botocore.exceptions.NoCredentialsError:
+            logger.error("AWS credentials not found!")
+            return False
+        except Exception as e:
+            if hasattr(e, 'response') and 'Error' in e.response:
+                code = e.response['Error']['Code']
+                if code == 'ResourceNotFoundException':
+                    if auto_create:
+                        logger.error(f"Remote state table {self.table_name} does not exist, creating...")
```
Comment (toVersus) [inline aws_eden_cli/dynamodb.py]:
```suggestion
                        logger.info(f"Remote state table {self.table_name} does not exist, creating...")
```

File: aws_eden_cli/cmdline.py [hunk 3]
```diff
@@ -194,107 +219,122 @@ def command_config_check(args):
...
-    if config is None:
-        return
-    config, _ = utils.config_write_overrides(args, config, args['profile'])
+def command_config_pull(args_dict: dict):
+    setup_logging(args_dict['verbose'])
+    profile_name = args_dict['profile']
 
-    table_name = args['remote_table_name']
-    table = dynamodb_resource.Table(table_name)
+    # args_dict['profile'] = 'default'
```
Comment (toVersus) [inline aws_eden_cli/cmdline.py]: Forgot to remove this line?

Comment (prog893) [reply aws_eden_cli/cmdline.py]: 😅

Comment (prog893) [reply aws_eden_cli/cmdline.py]: fixed in 1085552

**Review by toVersus — APPROVED** [review]
LGTM

**Pull request merged**